code/modules/mechs/components/_components.dm

```diff
@@ -56,7 +56,9 @@
 
 /obj/item/mech_component/proc/update_health()
 	total_damage = brute_damage + burn_damage
-	if(total_damage > max_damage) total_damage = max_damage
+//	if(total_damage > max_damage) //total_damage = max_damage
+	if(total_damage < 0) // Bandaid fix for mechs overhealing
+		total_damage = 0
```

Comment:
This was previously `if(total_damage < (max_damage - max_damage))`. I fat fingered the commit button with my suggestion.